connaisseur/flask_server.py

```diff
@@ -138,41 +137,37 @@ def __create_logging_msg(msg: str, **kwargs):
     return str({"message": msg, "context": dict(**kwargs)})
 
 
-def __create_json_patch(path: str, index: int, image: Image):
-    return {"op": "replace", "path": path.format(str(index)), "value": str(image)}
-
-
 def __admit(admission_request: AdmissionRequest):
-    policy = ImagePolicy()
-    parent_images = admission_request.wl_object.parent_images
-    patches = []
+    try:
+        logging_context = dict(admission_request.context)
+        policy = ImagePolicy()
+        parent_images = list(map(str, admission_request.wl_object.parent_images))
+        patches = []
 
-    for index, c_image in enumerate(admission_request.wl_object.container_images):
-        try:
-            logging_context = dict(admission_request.context, image=c_image)
+        for image in admission_request.wl_object.container_images:
+            logging_context.update(image=str(image))
 
             # child resources have mutated image names, as their parents got mutated
             # before their creation. this may result in mismatch of rules or duplicate
             # lookups for already approved images. so child resources are automatically
             # approved without further check ups, when their parents were approved
             # earlier.
-            if c_image in parent_images:
```

Comment:
Does this not work due to the image being an object? If so, can you not overwrite the equality for the object instead of transmuting the list of images to a list of strings and the image to a string?

Reply:
did as you asked